cmd/kube-controller-manager/app/core.go

```diff
@@ -138,8 +138,12 @@ func startRouteController(ctx ControllerContext) (bool, error) {
 		glog.Warning("configure-cloud-routes is set, but cloud provider does not support routes. Will not configure cloud provider routes.")
 		return false, nil
 	}
-	routeController := routecontroller.New(routes, ctx.ClientBuilder.ClientOrDie("route-controller"), ctx.InformerFactory.Core().V1().Nodes(), ctx.Options.ClusterName, clusterCIDR)
-	go routeController.Run(ctx.Stop, ctx.Options.RouteReconciliationPeriod.Duration)
```

Comment:
please leave the go routine launching statements separate... it makes it easier to reason about what happens synchronously, and what doesn't

Reply:
@liggitt  Thanks for quick reply,In the code I found some controller the go routine are separated, but others are write in combined，should separated them？